eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelData.scala

```diff
@@ -452,8 +451,13 @@ object RbfStatus {
 }
 
 sealed trait SpliceStatus
+sealed trait QuiescenceNegotiation extends SpliceStatus
 object SpliceStatus {
   case object NoSplice extends SpliceStatus
+  case class QuiescenceRequested(splice: CMD_SPLICE) extends QuiescenceNegotiation
+  case class InitiatorQuiescent(splice: CMD_SPLICE) extends QuiescenceNegotiation
+  case class ReceivedStfu(stfu: Stfu) extends QuiescenceNegotiation
+  case object NonInitiatorQuiescent extends QuiescenceNegotiation
```

Comment:
Just to make sure I understand that correctly, there are two distinct paths in the state machine depending on whether we're the initiator or not:

- if we're the initiator: `QuiescenceRequested -> InitiatorQuiescent -> NonInitiatorQuiescent`
- if we're not the initiator: `ReceivedStfu -> NonInitiatorQuiescent`

The non-initiator always sends its `stfu` after all of its pending updates have been signed, right? So when we receive `stfu`, we should verify that the sending side doesn't have any pending update. What about renaming `NonInitiatorQuiescent` to `QuiescenceReached` instead? Because at that point all updates in both directions have been signed?

Reply:
I had previously named `NonInitiatorQuiescent` to `Quiescent` (equivalent to `QuiescenceReached` but now the state machine goes like this:
 `QuiescenceRequested -> InitiatorQuiescent -> SpliceRequested` 

The last transition happens when the initiator receives `stfu` from the non-initiator.

Reply:
Right, that looks correct! It's even simpler than what I had in mind, this is better :+1: